nix/nginz.nix

```diff
@@ -0,0 +1,104 @@
+{ nginxModules
+, pkg-config
+, callPackage
+, stdenv
+, symlinkJoin
+, dockerTools
+, writeTextDir
+, runCommand
+, gnugrep
+, coreutils
+, which
+, nginx
+, inotify-tools
+, dumb-init
+, cacert
+, bashInteractive
+, openssl
+, openssl_1_1
+, lib
+, makeWrapper
+, writers
+}:
+let
+  zauth = callPackage ./pkgs/zauth { };
+
+  nginzModules = nginxModules // {
+    zauth = {
+      src = ../services/nginz/third_party/nginx-zauth-module;
+      inputs = [ pkg-config zauth ];
+    };
+  };
+
+  nginz = nginx.override {
+    openssl = openssl_1_1;
+    modules = [
+      nginzModules.vts
+      nginzModules.moreheaders
+      nginzModules.zauth
+    ];
+  };
+
+  reload-script = stdenv.mkDerivation {
```

Comment:
This name makes it look like it's an "additional side script" that runs and reloads nginz whenever the configuration changes.

But this is a wrapper we run, instead of running nginz directly.

I'd probably rename this `nginzWithReloader` or `nginzWrapped`, and have it write a `$out/bin/nginz` because we execute this as if it was `nginz`.

The fact that this is a shell script executing the real `nginz` is an implementation detail.

Reply:
> This name makes it look like it's an "additional side script" that runs and reloads nginz whenever the configuration changes.

That's exactly what that script does 😄  so maybe it does have a good name.

I can rename the variables here. I don't like the suggestion of `$out/bin/nginz` because that name has a meaning already inside the rest of wire-server, and it's a different name - locally we actually want to run nginz the binary, not the reload script. So we shouldn't introduce there name clashes that make it harder to talk about things.